Source/JavaScriptCore/dfg/DFGGraph.h

```diff
@@ -778,59 +788,43 @@ class Graph final : public virtual Scannable {
     {
         return NaturalBlockIterable(*this);
     }
-    
+
+    template<typename Functor>
+    class ForwardingFunc {
+    public:
+        explicit ForwardingFunc(const Functor& functor)
+            : m_functor(functor)
+        {
+        }
+
+        template<typename NodeType>
+        ALWAYS_INLINE auto operator()(NodeType*, Edge& edge) const -> decltype(std::declval<Functor>()(edge))
+        {
+            return m_functor(edge);
+        }
+
+    private:
+        const Functor& m_functor;
+    };
+
     template<typename ChildFunctor>
-    ALWAYS_INLINE void doToChildrenWithNode(Node* node, const ChildFunctor& functor)
+    ALWAYS_INLINE void doToChildren(Node* node, const ChildFunctor& functor)
     {
-        DFG_NODE_DO_TO_CHILDREN(*this, node, functor);
+        DFG_NODE_DO_TO_CHILDREN(*this, node, (ForwardingFunc<ChildFunctor>(functor)));
     }
     
     template<typename ChildFunctor>
-    ALWAYS_INLINE void doToChildren(Node* node, const ChildFunctor& functor)
+    ALWAYS_INLINE void doToAllChildren(Node* node, const ChildFunctor& functor)
     {
-        class ForwardingFunc {
-        public:
-            ForwardingFunc(const ChildFunctor& functor)
-                : m_functor(functor)
-            {
-            }
-            
-            // This is a manually written func because we want ALWAYS_INLINE.
-            ALWAYS_INLINE void operator()(Node*, Edge& edge) const
-            {
-                m_functor(edge);
-            }
-        
-        private:
-            const ChildFunctor& m_functor;
-        };
-    
-        doToChildrenWithNode(node, ForwardingFunc(functor));
+        DFG_NODE_DO_TO_ALL_CHILDREN_COMMON(*this, node, (ForwardingFunc<ChildFunctor>(functor)), APPLY_THING_TO_DO);
     }
 
     template<typename ChildFunctor>
-    ALWAYS_INLINE void doToChildrenWithCheck(Node* node, const ChildFunctor& functor)
+    ALWAYS_INLINE void doToChildrenBreakOnDone(Node* node, const ChildFunctor& functor)
```

Comment:
Ditto.

Reply:
It seems armv7 complains about the missing template arg. So, just use the current approach.